lib/ansible/modules/identity/cyberark/cyberark_authentication.py

```diff
@@ -0,0 +1,304 @@
+                "useSharedLogonAuthentication"]
+            headers["Authorization"] = cyberarkSession["token"]
+
+        if useSharedLogonAuthentication:
+            endPoint = "/PasswordVault/WebServices/auth/Shared/RestfulAuthenticationService.svc/Logoff"
+        else:
+            endPoint = "/PasswordVault/WebServices/auth/Cyberark/CyberArkAuthenticationService.svc/Logoff"
+
+    try:
+
+        response = open_url(
+            apiBaseUrl + endPoint,
+            method="POST",
+            headers=headers,
+            data=payload,
+            validate_certs=validate_certs)
+
+        result = None
+        changed = False
+
+        if response.getcode() == 200:
+
+            if state == "present":
+
+                if useSharedLogonAuthentication:
+                    token = json.loads(response.read())["LogonResult"]
+                else:
+                    token = json.loads(response.read())["CyberArkLogonResult"]
+
+                result = {
+                    "cyberarkSession": {
+                        "token": token,
+                        "apiBaseUrl": apiBaseUrl,
+                        "validate_certs": validate_certs,
+                        "useSharedLogonAuthentication":
+                        useSharedLogonAuthentication},
+                }
+
+                if newPassword is not None:
+                    changed = True
+
+            else:
+                result = {
+                    "cyberarkSession": {}
+                }
+
+            return (changed, result, response.getcode())
+        else:
+            module.fail_json(
+                msg="error in endpoint=>" +
+                endPoint +
+                json.dumps(headers))
+
+    except httplib.HTTPException:
+        t, e = sys.exc_info()[:2]
+        module.fail_json(
+            msg="EndPoint=" +
+            apiBaseUrl +
+            endPoint +
+            " headers=[" +
+            json.dumps(headers) +
+            "] ===>" +
+            str(e),
+            exception=traceback.format_exc(),
+            status_code=e.code)
+    except Exception:
+        t, e = sys.exc_info()[:2]
+        module.fail_json(
+            msg="EndPoint=" +
+            apiBaseUrl +
+            endPoint +
+            " headers=[" +
+            json.dumps(headers) +
+            "] ===>" +
+            str(e),
+            exception=traceback.format_exc(),
+            status_code=-1)
+
+
+def main():
+
+    fields = {
+        "apiBaseUrl": {"required": False, "type": "str"},
+        "validate_certs": {"required": False, "type": "bool",
+                          "default": "true"},
+        "username": {"required": False, "type": "str"},
+        "password": {"required": False, "type": "str", "no_log": True},
+        "newPassword": {"required": False, "type": "str", "no_log": True},
+        "useSharedLogonAuthentication": {"default": False, "type": "bool"},
+        "useRadiusAuthentication": {"default": False, "type": "bool"},
```

Comment:
It may be better to define only one required authentication param like authentication_method with choice between shared and radius
Then add a line in required_if for username and password in case of shared auth.